test/test_test.go

```diff
@@ -33,6 +33,56 @@ func TestAssertCanvasTappableAt(t *testing.T) {
 	assert.True(t, tt.Failed(), "test failed")
 }
 
+func TestAssertObjectRendersToImage(t *testing.T) {
+	obj := canvas.NewCircle(color.Black)
+	obj.Resize(fyne.NewSize(20, 20))
+
+	test.AssertObjectRendersToImage(t, "circle.png", obj)
+}
+
+func TestAssertObjectRendersToMarkup(t *testing.T) {
+	obj := canvas.NewCircle(color.Black)
+	obj.Resize(fyne.NewSize(20, 20))
+
+	assert.True(t, test.AssertObjectRendersToMarkup(t, "circle.xml", obj), "existing master is equal to rendered markup")
+}
+
+func TestAssertRendersToImage(t *testing.T) {
+	c := test.NewCanvas() // no painter, would be cycle - images will be blank
+	c.SetContent(canvas.NewCircle(color.Black))
+	c.Resize(fyne.NewSize(10, 10))
+
+	t.Run("non-existing master", func(t *testing.T) {
+		tt := &testing.T{}
+		assert.False(t, test.AssertRendersToImage(tt, "non_existing_master.png", c), "non existing master is not equal to rendered image")
+		assert.True(t, tt.Failed(), "test failed")
+		_, err := os.Stat("testdata/failed/non_existing_master.xml")
+		exists := os.IsNotExist(err)
```

Comment:
From the documentation of `IsNotExist`:
> New code should use errors.Is(err, fs.ErrNotExist).

Do we need to stick to `os.IsNotExist` to support older Go versions? AFAICS, `errors.Is` was added in Go 1.13.

Reply:
Oh, and the outcome of `os.IsNotExist(err)` is if the file does _not_ exist. Therefore this test seems to be a false positive.

Reply:
We don't need to - I just thought it read more cleanly. Wasn't the `fs.ErrNotExist` they recommend using not more recent than 1.14 though?

Reply:
But the test runs say it’s okay, what am I missing?

Reply:
> Wasn't the fs.ErrNotExist they recommend using not more recent than 1.14 though?

Probably not, `os.IsNotExist` checks for `fs.ErrNotExist`.

Reply:
It will in a modern package, but was the `io/fs` not new in 1.16?